### PR TITLE
Multipart mime 3487 v20

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/3487

Describe changes:
- convert HTTP to use new rust mime parser
- convert SMTP to use new rust mime parser

Follows #9436 with fix for not taking the en d of line before the boundary as part of the file

```
SV_BRANCH=pr/1371
```
SV tests from unit tests in https://github.com/OISF/suricata-verify/pull/1371

Draft : 
- I think #9435 should get merged first (and QA rebased over it)

### CODE REVIEW

**Comment by suricata-qa** [comment]
ERROR:
 ERROR: panic in ASAN_TLPR1_suri QA test

ERROR: QA failed on ASAN_TLPR1_suri.
  

Pipeline 15822

File: rust/src/mime/smtp.rs [hunk 1]
```diff
@@ -0,0 +1,825 @@
...
+        sbcfg,
+        md5: Md5::new(),
+        md5_state: MimeSmtpMd5State::MimeSmtpMd5Disabled,
+        md5_result: [0; 16].into(),
+    };
+    return Some(r);
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn rs_mime_smtp_state_init(
```
Comment (jasonish) [inline rust/src/mime/smtp.rs:143]: Public C API so should follow those naming conventions prefixed with `SC`.  See `rust/src/ffi/hashing.rs` as an example.

Comment (catenacyber) [reply rust/src/mime/smtp.rs:143]: For all `git grep 'pub unsafe extern "C"' rust/src/mime/` ?

Only for new ones ?
(I see that http2 for instance has the same `rs_http_parse_content_range`)

**Comment by catenacyber** [comment]
Replaced by https://github.com/OISF/suricata/pull/9443

File: rust/src/mime/smtp.rs (hunk 1 above)
Comment (catenacyber) [reply rust/src/mime/smtp.rs:143]: Commit pushed in my branch multipart-mime-3487-v23 ;-)

**Pull request closed without merging**